backend/connectors/databases_population.py

```diff
@@ -22,3 +24,9 @@
     # print data stored in the collection
     for item in my_collection.find():
         print(item)
+
+postgres_instance = PostgreSqlConnector()
+# MB: Connection string for a database created locally on my machine. PostgreSQL database cannot be created in memory
+# as far as I know
+with postgres_instance.connector(host="localhost", username="postgres", password="admin", db_name="pa036"):
+    pass
```

Comment:
Queries will be done in this 'with' clause?

Reply:
Connection has to be closed either by using "with" or by calling .close() method. Furthermore, if you want to launch query, you have to create cursor from connection by calling connection.cursor(). Cursor has to be then closed as well.

Reply:
This is an example from my diploma thesis:

![image](https://user-images.githubusercontent.com/38705514/116204418-0b569780-a73d-11eb-859d-fd3373dbd6d9.png)

Reply:
Actually the 'close()' does not have to be explicit anywhere, because of the connector interface's __exit__ method and the usage of 'with'